### DISCUSSION TITLE
Azure: Why use the service-bus trigger over azure monitor trigger?

### DISCUSSION THREAD

**Discussion by wkalt** [kedacore/keda#3150] [General] [answered]
Hi,
New user to KEDA and Azure here.

It seems to me that you could implement servicebus queue-based autoscaling in Azure using KEDA with either the service-bus trigger or the azure-monitor trigger. I'm trying to understand the tradeoffs. It seems like the service-bus trigger requires send/receive on the topic/queue (via the manage permission), while the azure-monitor trigger requires just read permission for the metric (docs actually say "resource" - is that the metric, or the topic?) - which seems more appropriate.

However, most of the examples I've found around the net are for the service-bus trigger. Is there some reason people use it instead, or something it does that the azure-monitor one does not?

**Comment by tomkerkhove** [comment, marked as answer]
Yes, Service Bus will scale faster given it's closely integrated with the service and Azure Monitor metrics can have a ~5 min delay.

**Reply by wkalt** [reply]
thanks!
